node/src/components/consensus/era_supervisor.rs

```diff
@@ -723,32 +599,38 @@ where
         }
     }
 
-    pub(super) fn handle_new_block_payload(
+    pub(super) fn handle_new_block_payload<REv: ReactorEventT<I>>(
         &mut self,
+        effect_builder: EffectBuilder<REv>,
+        rng: &mut NodeRng,
         new_block_payload: NewBlockPayload,
     ) -> Effects<Event<I>> {
         let NewBlockPayload {
             era_id,
             block_payload,
             block_context,
         } = new_block_payload;
-        if !self.era_supervisor.is_bonded(era_id) {
+        if !self.active_eras.contains_key(&era_id) || era_id.successor() < self.current_era {
```

Comment:
If we switched places of these statements, then the check could be slightly cheaper as comparing integers will be faster than checking set membership.

Reply:
Done in https://github.com/casper-network/casper-node/pull/2127/commits/f417ff94706385eda1d3d134cadcc7fec3f6dfd0.